store/sqlstore/post_store.go

```diff
@@ -1407,6 +1407,13 @@ func (s *SqlPostStore) getPostsAround(before bool, options model.GetPostsOptions
 		sq.Eq{"p.ChannelId": options.ChannelId},
 		sq.Eq{"p.DeleteAt": int(0)},
 	}
+
+	if options.IncludeDeleted {
+		replyCountSubQuery = s.getQueryBuilder().Select("COUNT(*)").From("Posts").Where(sq.Expr("Posts.RootId = (CASE WHEN p.RootId = '' THEN p.Id ELSE p.RootId END)"))
+		deleteAtIndex := len(conditions) - 1
+		conditions = conditions[:deleteAtIndex]
```

Comment:
We're relying on the `sq.Eq{"p.DeleteAt": int(0)},` condition to always be the last one. Someone modifying `conditions` could miss this. 

I suggest redefining `conditions` the way you did for `replyCountSubQuery`.


If you're up for it, I would suggest refactoring `replyCountSubquery` and `conditions` using the `queryBuilder` so that the `deletedAt` can we added conditionally instead of redefining them both, but this could end up being a bit complicated because of the `CASE` in the sql statement.